### DISCUSSION TITLE
presentation.Save() exception

### DISCUSSION THREAD

**Discussion by kaitenus** [ShapeCrawler/ShapeCrawler#617] [Q&A]
Hello,
I am trying some simple code to test write text to presentation:

var pres = new Presentation(@"D:\Report.pptx");
var slide = pres.Slides.First();

foreach (IShape shape in slide.Shapes) 
{
    if (shape.Name == "Head")
    {
        var table = (ITable)shape;
        table.Rows[0].Cells[1].TextFrame.Text = "TEST";
    }
}

pres.Save();

Exception is thrown at the code end -> pres.Save() -> Exception: Memory stream is not expandable.

Please help, thanks.

Best regards, Ivan

**Comment by AdamShakhabov** [comment]
Hello Ivan,

I am unable to reproduce this exception issue. Could you please attach your presentation?

If the presentation contains sensitive data and cannot be publicly shared, you can send it to [theadamo86@gmail.com](mailto:theadamo86@gmail.com).

**Comment by BiDuang** [comment]
I noticed that the `PresentationCore` use `new MemoryStream(bytes)` to init the Presentation from file path. This method may cause `not expandable stream` issue.

**Comment by sergedc** [comment]
I get the same error with this code:
            var source = new Presentation("D:\\DInsights\\Data access\\Analytics Model\\SlideLibrary.pptx");
            var target = new Presentation("D:\\DInsights\\Data access\\Analytics Model\\Addintest.pptx");
            var copyingSlide = source.Slides[1];
            target.Slides.Add(copyingSlide);
            target.Save();
            
  on the line target.Slides.Add(copyingSlide);
  
  This happens on both the main build and the prerelease

**Reply by BiDuang** [reply]
The #630 is now focusing to slove this problem, if you're available, please pull this code for testing.

**Reply by sergedc** [reply]
I am not the most familiar with pulling unpublished code. I am using Visual Studio (not code) and important ShapeCrawler using the "Manage nuget packages" option on my project. 
Could you tell me how to "pull this code"? A thousand thanks!

**Reply by BiDuang** [reply]
First, clone the ShapeCrawler repository and checkout the fixes-#617 branch:
   ```bash
   git clone https://github.com/BiDuang/ShapeCrawler
   cd ShapeCrawler
   git checkout fixes-#617
   ```
Open your **OWN** project in Visual Studio.

Right-click your solution in Solution Explorer and select `Add` -> `Existing Project...`.

Navigate to the location where you cloned the ShapeCrawler repository, then go to `ShapeCrawler\src\ShapeCrawler` and select `ShapeCrawler.csproj` to add the project to your solution.

After adding the project, your Solution Explorer should look similar to this:
   ![Solution Explorer](https://github.com/ShapeCrawler/ShapeCrawler/assets/40264028/8fd244fc-e8e8-45b6-a010-159c4e1994b0.png)

Next, open your project file (`.csproj`) in a text editor.
Add the following code inside the `<Project>` element:
   ```xml
   <ItemGroup>
     <ProjectReference Include="..\ShapeCrawler\src\ShapeCrawler\ShapeCrawler.csproj" />
   </ItemGroup>
   ```
Save the changes to your project file.

After following these steps, you should be able to use the ShapeCrawler project from the source code in your own project.

**Reply by sergedc** [reply]
Thank you very much for the explanation. 
By forcing me to do this I then faced a new problem: my project was a .net framework 4.8 while the ShapeCrawler was .net 8.0. I then learned about the difference and recreated the project as a .net 8.0 (ie not selection .net framework) and all problems went away (ie I did not need to use the 617 branch)

**Reply by AdamShakhabov** [reply]
@sergedc, does the latest version work for your project with target .NET Framework 4.8?

**Comment by kaitenus** [comment]
It's working. Thanks
